### PR TITLE
[16.0][MIG] product_uom_updatable

### PR DESCRIPTION
**Pull request by glitchov**
Standard migration from 14.0

### CODE REVIEW

File: product_uom_updatable/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        res = super().write(vals)
+        return res
+
+    def _update_uom(self, uom_id, field_name):
+        uom_obj = self.env["uom.uom"]
+        sorted_items = sorted(self, key=lambda r: r[field_name])
+        for key, products_group in groupby(sorted_items, key=lambda r: r[field_name]):
+            products = reduce(
+                lambda x, y: x | y, products_group, self.env["product.template"]
+            )
```
Comment (lmignon) [inline product_uom_updatable/models/product_template.py]: @glitchov It's not a good idea to use the operator '|' to merge list of recordset. It's less performant than the previous implementation ` product_ids = [product.id for product in products_group]` and far more difficult to read.

**Review by lmignon — CHANGES_REQUESTED** [review]

File: product_uom_updatable/models/product_template.py (hunk 1 above)
Comment (glitchov) [reply product_uom_updatable/models/product_template.py]: I need the recordset to invalidate after the update and i thought it was easier to get the ids from recordset than having the ids and re-browse for recordset.

**Review by lmignon — APPROVED** [review]
Thank you @glitchov   LGTM (Code review)

**Review by jbaudoux — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**